Add ability to easily roll spell focus checks

ashleytowner opened the pull request:
There was a [message in the discord](https://discordapp.com/channels/558029475837902851/1119133303514071090/1356773996011589722) asking about adding spell focus rolls so that the spell checks could be rolled without forcing the loss of a spell on a failed focus check. This PR adds a button to both PC and NPC sheets to do that. 

Considerations:

- Critical failures still result in the loss of a spell (as per rules-as-written)
- This code works primarily by hooking into pre-existing functions to cast spells, with minor changes in logic if it is a focus roll. I did this to reduce code duplication, but let me know if the preference would be for brand new functions. 

## Screenshots
![image](https://github.com/user-attachments/assets/3f6b0265-fcf8-4fa8-bdb2-5c7496fa38e2)
*Note that the button only appears on spells with a duration of "Focus"*

![image](https://github.com/user-attachments/assets/e8e04eaa-5e4b-4b51-83f2-7177c88d8670)
*A successful spell focus check*

![image](https://github.com/user-attachments/assets/18aef832-0cb9-4e51-ab5d-6ce12c11a31a)
*A failed spell focus check*

![image](https://github.com/user-attachments/assets/47c669b7-3eef-4ea5-ba7d-8c975266335b)
*A PC with both scrolls & wands*

PrototypeESBU commented:
I noticed that focus is not available on wands or scrolls. Not sure how scrolls would be tackled though.

Also the css changes to .actions affect a number of other tables negatively. Ideally actions should have a width of 75px for the spell table only.

ashleytowner commented:
> I noticed that focus is not available on wands or scrolls. Not sure how scrolls would be tackled though.
> 
> Also the css changes to .actions affect a number of other tables negatively. Ideally actions should have a width of 75px for the spell table only.

@PrototypeESBU I've fixed both of these issues. I've just added the same focus button to scrolls but I don't really know the best behaviour there.

PrototypeESBU commented:
Looks good. @Muttley Good to go on this one.
